### PR TITLE
Feature - email styles dark theme

### PR DESCRIPTION
**Pull request by jeffreyscottfrench**
### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:
Adds styles for dark theme / Dark Mode to all emails - email-styles.php
Adds fields in WC / Settings / Email for discrete dark theme styles, and dark theme alternate image - class-wc-settings-emails.php
Adds meta tags to WC Email Header for supporting light and dark themes - email-header.php

### How to test the changes in this Pull Request:

1. Email template preview via WC / Settings / Email honors adherence to system light/dark mode.
2.
3.

### Other information:

* [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [ ] Have you successfully run tests with your changes locally?

### Changelog entry
Feature - Adds settings and styles for dark theme support when sending WC Emails where supported by the user's email client.

> Enter a summary of all changes on this Pull Request. This will appear in the changelog if accepted.

### CODE REVIEW

File: includes/admin/settings/class-wc-settings-emails.php [hunk 1]
```diff
@@ -125,10 +125,22 @@ public function get_settings() {
...
+					'default'     => '',
+					'autoload'    => false,
+					'desc_tip'    => true,
+				),
+
 				array(
 					'title'       => __( 'Footer text', 'woocommerce' ),
 					/* translators: %s: Available placeholders for use */
-					'desc'        => __( 'The text to appear in the footer of all WooCommerce emails.', 'woocommerce' ) . ' ' . sprintf( __( 'Available placeholders: %s', 'woocommerce' ), '{site_title} {site_url}' ),
+					'desc'        => __( 'The text to appear in the footer of all WooCommerce emails.', 'woocommerce' ) . ' ' . sprintf( __( 'Available placeholders: %s', 'woocommerce' ), '{site_title} {site_address}' ),
```
Comment (claudiosanches) [inline includes/admin/settings/class-wc-settings-emails.php:143]: It introduces a regression that doesn't sounds related to this PR.

**Review by claudiosanches — CHANGES_REQUESTED** [review]

File: includes/admin/settings/class-wc-settings-emails.php (hunk 1 above)
Comment (jeffreyscottfrench) [reply includes/admin/settings/class-wc-settings-emails.php:143]: Looks like this was changed in #25630 which was included in the 4.2.0 release that was published after this pull request was submitted. I will update and resubmit.

Comment (jeffreyscottfrench) [reply includes/admin/settings/class-wc-settings-emails.php]: Fixed on this PR.

**Comment by claudiosanches** [comment]
Seems like you need to rebase your PR, it's now showing other commits from master.

**Comment by jeffreyscottfrench** [comment]
Rebased to tag 4.2.1 - if thats not where it should be branched from, let me know.

**Comment by jeffreyscottfrench** [comment]
Created alternate PR #27114 for same feature which more closely follows contribution guidelines.

**Pull request closed without merging**